src/Bundle/Config/ConfigResolver.php

```diff
@@ -91,7 +91,8 @@ private function buildLoadingOrder(): array
                 $loadingOrder[$name][] = $package;
             }
         }
-
+        ksort($loadingOrder);
```

Comment:
```suggestion
        uksort($loadingOrder, function (string $a, string $b): int {
            return md5($a) <=> md5($b);
        });
```

Reply:
That is not deterministic anymore as `md5` might have collisions. :smile_cat:

Reply:
So now if I want to be first, I have to name my extension `j4qxp8` instead of `aaa_something`? Because the MD5 hash of `j4qxp8` is `05c28b9764f3e912c55b38fba3aafa97`. 🙈

Reply:
That would be the outcome I already mentioned on mumble and therefore I refrained from implementing this.
To be honest, my concern is that the output of the current system is unreliable and I try to fix this behaviour.

Making the life of developers hard via "You must not name your bundle so that you can predict the loading order via name" is beyond the scope of this PR.
This is already explicitely not supported as the loading order can only be noted via load after.
Loading a bundle explicitly before another one is not supported since Contao 4.0 pre discussions during Ahrweiler 1.
Implementing a hash like md5 or the like will shift the problem only instead of solving it.

My only concern was, that the current implementations output is unpredictable and I solved it.

If you really insist on making the order unpredictable, feel free to reverse the sorting order randomly in patch releases, as this is the most sufficient solution to this problem while maintaining reproducible results for a given version of Contao. :smirk:

Reply:
I think most developers that know how to get an md5 which starts with `0` while using a sane name understand that they shouldn’t do that. But we could easily prevent that with something like that:

```suggestion
        ksort($loadingOrder);
        $salt = md5(implode("\0", array_keys($loadingOrder)));

        uksort($loadingOrder, function (string $a, string $b) use($salt): int {
            return md5($a.$salt) <=> md5($b.$salt);
        });
```

But I think this is not necessary, so I would go with the simple sort by md5 hash from above.

Reply:
@ausi that would change the loading order when you install a new bundle, wouldn't it? That should be prevented imho, otherwise it's bad for the user experience in the back end again.

Reply:
> that would change the loading order when you install a new bundle, wouldn't it?

Not if we follow my last sentence in the comment 🙃 
> However I think this is not necessary, so I would go with the simple sort by md5 hash from above.

@aschempp Can you elaborate on the thumbs-down? Do you want to use the simple md5 version or do you want to sort it alphabetically, or something else?

Reply:
> Not if we follow my last sentence in the comment 🙃

Well yes, but then someone could influence the loading order by naming the bundle in a certain way (though I agree this is a negligible issue…).

Reply:
The simple `md5`ed sort has the benefit, that the same two bundles are always sorted the same way relative to each other, in every installation, everywhere. I think it is the perfect tradeof between stability/reproducibility and preventing bundle-name misuse.